pgjdbc/src/main/java/com/yugabyte/ysql/ClusterAwareLoadBalancer.java

```diff
@@ -32,66 +36,90 @@ public int getRefreshListSeconds() {
 
   protected int refreshListSeconds = LoadBalanceProperties.DEFAULT_REFRESH_INTERVAL;
 
-  public ClusterAwareLoadBalancer() {
+  public ClusterAwareLoadBalancer(LoadBalanceService.LoadBalance lb, int refreshInterval) {
+    this.loadBalance = lb;
+    this.refreshListSeconds = refreshInterval;
   }
 
-  public static ClusterAwareLoadBalancer getInstance(int refreshListSeconds) {
+  public static ClusterAwareLoadBalancer getInstance(LoadBalanceService.LoadBalance lb,
+      int refreshListSeconds) {
     if (instance == null) {
       synchronized (ClusterAwareLoadBalancer.class) {
         if (instance == null) {
-          instance = new ClusterAwareLoadBalancer();
+          instance = new ClusterAwareLoadBalancer(lb, refreshListSeconds);
           instance.refreshListSeconds =
               refreshListSeconds >= 0 && refreshListSeconds <= LoadBalanceProperties.MAX_REFRESH_INTERVAL ?
                   refreshListSeconds : LoadBalanceProperties.DEFAULT_REFRESH_INTERVAL;
-          LOGGER.fine("Created a new cluster-aware LB instance with refresh" +
-              " interval " + instance.refreshListSeconds + " seconds");
+          LOGGER.fine("Created a new cluster-aware LB instance with loadbalance = " +
+              instance.loadBalance + " and refresh interval " + instance.refreshListSeconds + " seconds");
         }
       }
     }
     return instance;
   }
 
+  public String toString() {
+    return this.getClass().getSimpleName() + ": loadBalance = " + loadBalance + ", refreshInterval = " + refreshListSeconds;
+  }
+
   @Override
-  public boolean isHostEligible(Map.Entry<String, LoadBalanceService.NodeInfo> e) {
-    return !attempted.contains(e.getKey()) && !e.getValue().isDown();
+  public boolean isHostEligible(Map.Entry<String, LoadBalanceService.NodeInfo> e,
+      Byte requestFlags) {
+    return !attempted.contains(e.getKey()) && !e.getValue().isDown()
```

Comment:
A minor suggestion if you agree:
use a local variable and name appropriately to suggest what exactly is in the key.

`String host = e.getKey();`